### DISCUSSION TITLE
Creating and Customizing a Time Series Model

### DISCUSSION THREAD

**Discussion by ngupta23** [pycaret/pycaret#1757] [FAQs]
# Create a Time Series Model

```python
#### Install library (one time) ----
!pip install pycaret-ts-alpha -U 

#### Load data ----
from pycaret.datasets import get_data
y = get_data("airline")

#### Setup experiment ----
from pycaret.internal.pycaret_experiment import TimeSeriesExperiment
exp = TimeSeriesExperiment()
exp.setup(data=y, fh=12, session_id=42)

##### List the available models ----
exp.models()

#### Create a model from available models ----
model = exp.create_model("arima")
```

![image](https://user-images.githubusercontent.com/33585645/138568399-c035848a-0071-4787-ae9b-8d05ed453579.png)

* By default `pycaret` will perform 3 fold cross validation with an expanding/rolling window (see #1761).
* The display shows the metrics for all 3 folds along with the mean and standard deviation of the results.

# Model Details

Note that all models in the `pycaret` time series module are based on [`sktime`](https://github.com/alan-turing-institute/sktime). By default, the models that are created in `pycaret` will have the default arguments (with some exceptions - more on that in another FAQ)

```python
#### What model is this? ----
type(model)
model
```
```python
sktime.forecasting.arima.ARIMA

ARIMA(maxiter=50, method='lbfgs', order=(1, 0, 0), out_of_sample_size=0,
      scoring='mse', scoring_args=None, seasonal_order=(0, 1, 0, 12),
      start_params=None, suppress_warnings=False, trend=None,
      with_intercept=True)
```

# Customizing Models

While the default model is usually a good starting point, experienced modelers may want to perform some exploratory analysis on the dateset and set specific parameters in the model. This can be easily achieved by passing `kwargs` to the `create_model()` call. Note that the modeler can set any model parameter that is available in the `sktime` forecaster. In the example below, we show setting of the autoregressive `order` and the seasonal parameter `seasonal_order`.

```python
#### Create a better model (hand tuned) ----
better_model = exp.create_model("arima", order=(0, 0, 0), seasonal_order=(1, 1, 0, 12))
better_model
```

```python
ARIMA(maxiter=50, method='lbfgs', order=(0, 0, 0), out_of_sample_size=0,
      scoring='mse', scoring_args=None, seasonal_order=(1, 1, 0, 12),
      start_params=None, suppress_warnings=False, trend=None,
      with_intercept=True)
```

![image](https://user-images.githubusercontent.com/33585645/138568498-7a1c0fc2-ac9e-4981-96b2-6b068cf55e44.png)

**Comment by moezali1** [comment]
PyCaret 4.0 sunset the legacy time-series module (sktime/prophet integration), so this FAQ no longer maps to a supported code path.

If you still need time-series forecasting, use sktime / prophet / statsforecast directly — those projects have surpassed what the PyCaret wrapper offered.

For 4.0 docs, see https://pycaret.org/docs and the cleanup notice at https://github.com/pycaret/pycaret/issues/4177.

Closing as outdated.
